src/realm/object-store/shared_realm.cpp

```diff
@@ -910,6 +910,22 @@ void Realm::close()
     m_coordinator = nullptr;
 }
 
+bool Realm::delete_files()
+{
+    if (!is_closed()) {
+        return false;
+    }
+
+    return DB::call_with_lock(m_config.path, [&](auto) {
+        util::File::try_remove(m_config.path);
+        util::File::try_remove(m_config.path + ".log");
+        util::File::try_remove(m_config.path + ".log_a");
+        util::File::try_remove(m_config.path + ".log_b");
+        util::File::try_remove(m_config.path + ".note");
+        util::try_remove_dir_recursive(m_config.path + ".management");
```

Comment:
Thank you, this is exactly what I was looking for!

It does lead to a bunch of very interesting questions first though:
- https://github.com/realm/realm-core/blob/master/src/realm/db.hpp#L360-L365:
`The temporary files are not returned by this function.` 
Assuming that temporary == optional I would suggest to introduce a parameter indicating if optional files should be included or not:
`static std::vector<std::pair<std::string, bool>> get_core_files(const std::string& realm_path, bool should_include_optional_files);`
This way `common_dir.cpp` (https://github.com/realm/realm-core/blob/master/src/realm/sync/noinst/common_dir.cpp#L7) which is the only caller at the moment could function unchanged.

- This would still mean that we use the extension strings in several locations (implementation + tests). Another suggestion could be:
```
enum CoreFileExtensionType {
    Log, LogA, LogB, Note, Management, Backup, Lock
};
static std::map<DB::CoreFileExtensionType, std::pair<std::string, bool>> get_core_file_extensions();
```
where the returned pair works like the one on `get_core_files()` just with the extension instead of the whole path. And a bool indicating a folder.
or
```
enum CoreFileExtensionType {
    Log, LogA, LogB, Note, Management, Backup, Lock
};
static std::pair<std::string, bool> get_core_file_extension(DB::CoreFileExtensionType file_extension_Type);
```
as above but providing a specific extension type.

One of those then used by `get_core_files()` or just directly by the caller of `get_core_files()`.

- In doing so we could also just implement `delete_files()` in `db.cpp` instead of `shared_realm.cpp` to have just one implementation for both ways and then expose this function via `common_dir.hpp` for Sync and `shared_realm.hpp` for the SDKs. Alternative: import `db.hpp` in `server.cpp` which is the only user of `remove_realm_file()` from `common_dir.hpp` anyway.

Thoughts on that? :)

- Since we talked about the `.lock` yesteday a lot: Is https://github.com/realm/realm-core/blob/master/src/realm/sync/noinst/common_dir.cpp#L17 in fact unsafe then and should be changed as well?

Reply:
To provide for flexibility, as well as maintaining backwards compatibility I suggest this API:

```
enum CoreFileExtensionType : uint64_t {
    Log = 1,
    LogA = 2,
    LogB = 4,
    Note = 8,
    Management = 16,
    Storage = 32,
    Backup = 64,
    Lock = 128,

    TemporaryFiles = Log | LogA | LogB | Note,
    StateFiles = Management | Storage,
};
static std::pair<std::string, bool> get_core_files(uint64_t type = CoreFileExtensionType::StateFiles);
```

That should allow you to keep the existing callers unchanged without breaking anything, while also allowing you to specify exactly which files you want from the new call site.

Reply:
https://github.com/realm/realm-core/blob/master/src/realm/sync/noinst/common_dir.cpp#L17 is only allowed because the sync server assumes that it has complete and exclusive control over it's copy of the server side sync Realm files. It would be unsafe if people were messing around in the server files, but it was reasonable to assume people wouldn't do that. This server implementation is only used in our test sync implementation now as well so shouldn't be a problem, but good eye.

Reply:
Added f725bc91be706b85e1a5d6a53e1b78b0fbab716f and f9f9101d60b36050699a4afa78471c7ccc776cac to address this. Looking forward to your thoughts.